_posts/2017-01-09-migrating-to-puppet4.md

```diff
@@ -0,0 +1,26 @@
+---
+layout: post
+title: Migrating a Module to Puppet 4
+date: 2017-01-09
+github_username: bastelfreak
+twitter_username: BastelsBlog
+---
+
+
+We announced the deprecation of Puppet 3 in our modules
+[already](https://voxpupuli.org/blog/2016/12/22/putting-down-puppet-3/). There
+are a few steps needed to be taken before a Puppet-4-only release can be done:
+
+* Check the current metadata.json for the required Puppet version and bump it if needed
+  * We already have a few modules that only support Puppet 4, in this case you can ignore this guide
+  * The minimal required Puppet Version should not be 4.0.0 but [4.6.1](https://github.com/voxpupuli/community-triage/blob/master/modules/notes/2017-01-05.md#discussion)
+  * Also check if the dependencies in the `metadata,json` all have Puppet 4 support, probably they have to be bumped too
+  * The minimal required stdlib version has to be 4.6.0
+* Do a last release with Puppet 3 support which announces the next major release without Puppet 3
```

Comment:
How about `Perform a release of the module within the current Major version that announces the deprecation of Puppet 3. e.g. Current version is 2.1.3, release 2.2.0 with the deprecation notice`